### DISCUSSION TITLE
Clear window assignment to workspace

### DISCUSSION THREAD

**Discussion by ghost** [i3/i3#5706] [Q&A]
Hello!

I have what I assume to be a somewhat common use-case in which I assign applications to workspaces so that they will spawn in their designated workspace when lanunched. The issue I am facing is that I'd like additional windows for programs to spawn in the active workspace (eg Firefox should always start in Workspace 1, but if I were to move it to a different ws and start a new window with Ctrl+N, that window should appear in the active workspace, instead of on ws1 as it would do under a regular assignment rule). I understand that this distinction is beyond the scope of what i3 sets out to accomplish, so my next idea was to apply some logic on top of my executions, in such a way that:
- I could read the class of the parent PID for the new window to determine if it should move or stay, or
- I could do a normal assignment from either their desktop entries, or my startup script.
I prefer the second option, however that would require me to find the opposite of `assign <criteria> [→] [workspace] [number] <workspace>` , to restore the default behavior to newly spawned windows, and I have not found it in the docs or this sub.
Is anyone here aware of such an option? I have tried `assign [class="Firefox"] workspace current` to no avail.

Thanks!

**Comment by orestisfl** [comment]
It should be quite easy to watch for the window event using a script and if the class is Firefox and move the window to workspace 1 only if there is no other Firefox window in the current workspace.

See https://i3wm.org/docs/ipc.html
